Sources/Sentry/SentryTracer.m

```diff
@@ -217,14 +217,20 @@ - (void)dispatchIdleTimeout
         [self.dispatchQueueWrapper dispatchCancel:_idleTimeoutBlock];
     }
     __weak SentryTracer *weakSelf = self;
-    _idleTimeoutBlock = dispatch_block_create(0, ^{
+    _idleTimeoutBlock = [self.dispatchQueueWrapper createDispatchBlock:^{
         if (weakSelf == nil) {
             SENTRY_LOG_DEBUG(@"WeakSelf is nil. Not doing anything.");
             return;
         }
         [weakSelf finishInternal];
-    });
-    [self.dispatchQueueWrapper dispatchAfter:self.idleTimeout block:_idleTimeoutBlock];
+    }];
+    if (_idleTimeoutBlock == NULL) {
+        SENTRY_LOG_WARN(@"Couln't create idle time out block. Can't schedule idle timeout. "
+                        @"Finishing transaction");
+        [self finishInternal];
```

Comment:
m: I would just add a comment here that the transaction will not be send because it does not have any child.

Reply:
Better now?